### ISSUE TITLE
BitConverter.HalfToInt16Bits(Half) result comparison issue

### ISSUE THREAD

**Issue by gcsizmadia** [dotnet/runtime#82680]
### Description

I am playing around the negative zero number and I found an interesting thing while I used `BitConverter.HalfToInt16(Half)` method.

The two comparison gives different result
- Compare the method result directly to 0.
- Store the method result in a `short` variable then compare the variable to 0.

### Reproduction Steps

```csharp
Half half = (Half)(-0.0f);

// It will be -32768.
short bits = BitConverter.HalfToInt16Bits(half);

// It will be true.
bool isMethodResultGreaterThanOrEqualToZero = BitConverter.HalfToInt16Bits(half) >= 0;

// It will be false.
bool isVariableGreaterThanOrEqualToZero = bits >= 0;
```

### Expected behavior

The expected behavior is that the two comparison method should have the same result: `false`

### Actual behavior

- When directly check if the method result is `>= 0` then the comparison result will be `true`.
- When storing the method result first in a variable and then check if the variable is `>= 0` then the comparison result will be `false`.

### Configuration

- Which version of .NET is the code running on?
  `.NET 6.0.14` and `.NET 7.0.3`
- What OS and version, and what distro if applicable?
  `Windows 11` - `Microsoft Windows [Version 10.0.22621.1265]`
- What is the architecture (x64, x86, ARM, ARM64)?
  `x64`
- Do you know whether it is specific to that configuration?
  `I don't know.`

**Comment by Clockwork-Muse** [comment]
I'm getting false for both.  
Can you please write a small reproduction project that shows this issue?

**Comment by gcsizmadia** [comment]
It seems that this issue applies only to `.NET 6.0.14`. In `.NET 7.0.3` both values are `false`.

### .NET 6.0.14
![image](https://user-images.githubusercontent.com/8214318/221405277-74073920-ce44-4234-bbf2-cc6f53bbdacf.png)
![image](https://user-images.githubusercontent.com/8214318/221405308-61fccce0-420f-4640-a4c6-0cb03cdb0059.png)

### .NET 7.0.3
![image](https://user-images.githubusercontent.com/8214318/221405416-0e411dd4-3399-4136-8948-19076d1bd96e.png)

**Comment by gcsizmadia** [comment]
I attached a .zip file with a simple console app that targets both `.net6.0` and `.net7.0`.

[BitConverterTest.zip](https://github.com/dotnet/runtime/files/10833478/BitConverterTest.zip)

**Comment by KeterSCP** [comment]
It also has different output for Debug and Release in .NET 6.0.14

Debug:
```
-32768
True
False
```

Release:
```
32768
True
True
```

**Comment by JulieLeeMSFT** [comment]
@gcsizmadia, is there product impact with .NET 6.0.14? Can you work with .NET 7.0.3?

> It seems that this issue applies only to `.NET 6.0.14`.

**Comment by gcsizmadia** [comment]
@JulieLeeMSFT, I'm working on a class library (nuget package) that targets both `net6.0` and `net7.0` as well and its functions must work on both. So unless there is a workaround that can be applied to `net6.0` target, it is a product impact.

**Comment by jakobbotsch** [comment]
This sounds like the same issue as #61359 that was fixed in .NET 7+ by #64881. I think we can backport that fix to .NET 6.

**Comment by jakobbotsch** [comment]
This was backported and the fix is available in 6.0.16 that was released on April 11th.
